File: crates/facet_generate/src/tests/test_opaque_proxy_field/output.swift

/// FFI view. Field holds a lattice type. Serialized as `UserId`.
public struct SignInView {
    public var userId: UserId

    public init(userId: UserId) {
        self.userId = userId
    }
}

/// Domain ID with `Facet`. Wire anchor for typegen.
public struct UserId {
    public var value: String

    public init(value: String) {
        self.value = value
    }
}

/// Newtype wrapper around an opaque lattice field. Serialized as `UserId`.
public struct UserIdWrapper {
    public var value: UserId

    public init(value: UserId) {
        self.value = value
    }
}

File: crates/facet_generate/src/tests/test_opaque_proxy_field/output.ts
type str = string;

/// FFI view. Field holds a lattice type. Serialized as `UserId`.
export class SignInView {
    constructor (public user_id: UserId) {
    }
}

/// Domain ID with `Facet`. Wire anchor for typegen.
export class UserId {
    constructor (public value: str) {
    }
}

/// Newtype wrapper around an opaque lattice field. Serialized as `UserId`.
export class UserIdWrapper {
    constructor (public value: UserId) {
    }
}
